### PR TITLE
Bump HTTP compat to "1", drop support for Julia versions below 1.6

### PR DESCRIPTION
**Pull request by etpinard**
I went ahead and cherry-picked @mind6's commits from https://github.com/plotly/Dash.jl/pull/175 unto the current `dev` branch.

---

Since we had to disable the Percy tests (ref https://github.com/plotly/Dash.jl/issues/177), I would be nice to manually test out this branch before merging it. I'll personally do that starting now.

I'm planning on releasing Dash `v1.2.0` early next week.

### CODE REVIEW

File: src/server.jl [hunk 1]
```diff
@@ -52,7 +52,7 @@ function run_server(app::DashApp,
     start_server = () -> begin
         handler = make_handler(app);
         server = Sockets.listen(get_inetaddr(host, port))
-        task = @async HTTP.serve(handler, host, port; server = server, verbose = true)
+        task = @async HTTP.serve(handler, host, port; server = server, verbose = debug==true)
```
Comment (etpinard) [inline src/server.jl:55]: Personally I would set `verbose=false` even in under `run_server(app; debug=true)` 

I'd say logs like

```
┌ Debug: startread called
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:427
┌ Debug: invoking handler
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:446
┌ Debug: closeread
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:452
┌ Debug: closewrite
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:454
┌ Debug: startread called
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:427
┌ Debug: invoking handler
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:446
┌ Debug: closeread
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:452
┌ Debug: closewrite
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:454
┌ Debug: startread called
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:427
┌ Debug: invoking handler
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:446
┌ Debug: closeread
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:452
┌ Debug: closewrite
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:454
┌ Debug: startread called
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:427
┌ Debug: invoking handler
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:446
┌ Debug: closeread
└ @ HTTP.Servers ~/.julia/packages/HTTP/z8l0i/src/Servers.jl:452
```

are pretty much useless for Dash.jl users. Thoughts?

Comment (mind6) [reply src/server.jl:55]: yes I think it's overly verbose for Dash and useless

Comment (etpinard) [reply src/server.jl:55]: thanks for your input @mind6 !

**Comment by etpinard** [comment]
Ok. I'm going to go ahead and merge this and try to release Dash.jl `v1.2.0` - wish me luck :crossed_fingers:

**Comment by etpinard** [comment]
Now available as `v1.2.1` from Julia's general registry https://github.com/JuliaRegistries/General/pull/77586 :tada:

**Pull request merged**